docs/mcp/client.md

```diff
@@ -169,7 +169,39 @@ The MCP servers provide the ability to set a `process_tool_call` which allows
 the customisation of tool call requests and their responses.
 
 A common use case for this is to inject metadata to the requests which the server
-call needs.
+call needs. Accessing the metadata is then SDK specific, for example with the
```

Comment:
Thanks for adding het new section, can you add it below the existing example though? I think the flow of "here's how to send metadata", "here's how to receive it" is more clear.